modules/@apostrophecms/ui/ui/apos/stores/notification.js

```diff
@@ -0,0 +1,189 @@
+import { defineStore } from 'pinia';
+import { ref } from 'vue';
+
+export const useNotificationStore = defineStore('notification', () => {
+  const notifications = ref([]);
+  const dismissed = ref([]);
+  const processes = ref({});
+
+  /**
+   * @param {string} message - Notification message
+   * @param {object} options.interpolate - Translation interpolate object
+   * @param {string} options.type - Notification type
+   * @param {string} options.icon - Notification icon
+   * @param {boolean|number} options.dismiss - Notification dismiss behavior
+   * @param {array} options.buttons - Notification buttons
+   * @param {boolean} options.localize - Should the message be localized
+   * @returns string - Returns ID of the created notification
+   */
+  async function notify(message, options = {}) {
+    if (options.dismiss === true) {
+      options.dismiss = 5;
+    }
+
+    const notif = {
+      message,
+      interpolate: options.interpolate || {},
+      type: options.type,
+      icon: options.icon,
+      dismiss: options.dismiss,
+      buttons: options.buttons,
+      localize: options.localize
+    };
+
+    // Send it to the server, which will send it back to us via polling
+    const { noteId } = await apos.http.post(apos.notification.action, { body: notif });
+    return noteId;
+  }
+
+  /**
+   * @param {string} notifId - Notification ID
+   */
+  async function dismiss(notifId) {
+    await apos.http.patch(`${apos.notification.action}/${notifId}`, {
+      body: {
+        dismissed: true
+      }
+    });
+    notifications.value = notifications.value.filter(
+      ({ _id }) => notifId !== _id
+    );
+    if (processes.value[notifId]) {
+      delete processes.value[notifId];
+    }
+  }
+
+  async function poll() {
+    try {
+      if (document.visibilityState === 'hidden') {
+        // Wait for tab to become visible
+        setTimeout(poll, 5000);
+      } else {
+        const allNotifications = [ ...notifications.value, ...dismissed.value ];
+        const latestTimestamp = allNotifications
+          .map((notification) => {
+            return notification.updatedAt;
+          })
+          .sort()
+          .reverse()[0];
```

Comment:
```suggestion
          .sort()
          .at(-1);
```

does this work?

Reply:
Probably, not my code I just moved it, didn't want to bring to much changes..

Reply:
Updated